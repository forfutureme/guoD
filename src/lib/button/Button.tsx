/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-11 20:57:58
 * @Desc: Button 按钮
 */
import React from 'react'
import './button.scss'
import {prefixClass, joinClassName} from '@utils/classNameHandle'
const fixClass = prefixClass('guo-button')

interface props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  theme?: string
}

const Button: React.FC<props> = ({
  theme = 'default',
  className,
  children,
  onClick,
  ...rest
}) => {
  return (
    <button
      className={joinClassName(fixClass(), fixClass(theme), className)}
      onClick={onClick}
      {...rest}>
      {children}
    </button>
  )
}

export default Button
