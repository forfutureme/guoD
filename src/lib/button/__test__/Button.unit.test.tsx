/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-11 21:52:03
 * @Desc: 按钮 单元测试
 */
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'
import React from 'react'
import Button from '../Button'

describe('Button测试', () => {
  it('测试Button渲染', () => {
    const json = renderer.create(<Button />)
    expect(json).toMatchSnapshot()
  })
  it('测试Button点击事件', () => {
    const fn = jest.fn()
    const component = mount(<Button onClick={fn} />)
    component.find('button').simulate('click')
    expect(fn).toBeCalled()
  })
})
