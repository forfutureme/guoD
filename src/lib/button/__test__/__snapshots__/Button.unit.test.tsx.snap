// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button测试 测试Button渲染 1`] = `
<button
  className="guo-button guo-button-default"
/>
`;
